tests/unit-tests/conversions.cpp

```diff
@@ -59,4 +51,107 @@ TEST_CASE("C++ Standard Conversions") {
       lexicon.double_type()
   );
   lexicon.make_if_then(condition, *lexicon.make_expr_stmt(then_expr));
+
+  // Bitcast                      (Pretend)
+  INFO("(bool*)ptr");
+  lexicon.make_pretend(
+    *lexicon.make_id_expr(ptr),
+    lexicon.get_pointer(lexicon.bool_type()),
+    lexicon.get_pointer(lexicon.bool_type())
+  );
+}
+
+TEST_CASE("Class Conversions") {
+  using namespace ipr;
+  impl::Lexicon lexicon{};
+  impl::Module module{lexicon};
+  impl::Interface_unit unit{lexicon, module};
+
+  INFO("struct Base {};");
+  INFO("struct Derived : Base {};");
+  auto& base = *lexicon.make_class(*unit.global_region());
+  auto& derived = *lexicon.make_class(*unit.global_region());
+  derived.declare_base(base);
+
+  INFO("Base b;");
+  INFO("Derived d;");
+  auto& base_ptr = lexicon.get_pointer(base);
+  auto& derived_ptr = lexicon.get_pointer(derived);
+  auto& b = *unit.global_region()->declare_var(*lexicon.make_identifier("b"), base_ptr);
+  auto& d = *unit.global_region()->declare_var(*lexicon.make_identifier("d"), derived_ptr);
+
+  // Derived-to-base conversion
+  INFO("(Base*)b;");
+  lexicon.make_narrow(
```

Comment:
A narrow cast does not necessarily map to a C-cast, semantically.  I am not quite sure what the text pi `INFO` is supposed to mean here...

Reply:
These test cases are meant to show how to translate source code to it IPR representation. The info are simply and example snippet of source code to help illustrate to a user how they might go about doing so.

Reply:
In that case, I don't know we have the right notations.  The implicit conversions don't have syntax: They just happen in context, that is why they are implicit.  

Right now, I don't have a suggestion for "this is what happens in context"

Reply:
The Narrow node is the abstract concept of the conversion, it is implicitly applied in the language but it can also be performed explicitly.  I can't test every possible route that produces this abstract conversion but it may help to provide an implicit variant as well as this explicit variant so a reader is not thrown off when reading this test case.

Reply:
OK, understood